// vue-fixed-header/dist/vue-fixed-header.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
	typeof define === 'function' && define.amd ? define(factory) :
	(global.vueFixedHeader = factory());
}(this, (function () { 'use strict';

var $;
var tag;

function getTargetTag() {
  if (navigator.userAgent.includes('Edge')) { return 'body'; }
  if (!window.chrome && 'WebkitAppearance' in document.documentElement.style) { return 'body'; }
  return 'html';
}

var index = {
  name: 'fixed-header',

  props: {
    threshold: {
      type: Number,
      default: 0,
      required: false
    }
  },

  mounted: function mounted() {
    $ = function (e) { return document.querySelector(e); };
    tag = getTargetTag();
    this.main();
  },

  methods: {
    main: function main() {
      var this$1 = this;

      this.check = function () {
        this$1.$emit('update:fixed', $(tag).scrollTop > this$1.threshold);
      };
      window.addEventListener('scroll', this.check);
    }
  },

  beforeDestory: function beforeDestory() {
    window.removeEventListener('scroll', this.check);
  },

  render: function render(h) {
    var children = this.$slots.default;
    return children[0];
  }
};

return index;

})));
